### ISSUE TITLE
Unhelpful error message when running `dotnet add <PACKAGE_NAME>`

### ISSUE THREAD

**Issue by jpierson** [dotnet/cli#6926]
## Steps to reproduce

1. Ensure latest .NET CLI is installed (2.0.0-preview1-005977).
2. Open command line and navigate to directory that contains a .NET project (.fsproj in my case).
4. Run command missing the `package` "command" as such `dotnet add FSharp.Data.TypeProviders`.

## Expected  behavior

1. Inform the user specifically that with the `dotnet add` command that either *package* or *reference* must also be specified.
2. Detecting that `FSharp.Data.TypeProviders` is a valid available package and not a name of an available project and implying the behavior of `dotnet add package FSharp.Data.TypeProviders` or presenting the user with a "did you mean ..." experience.

## Actual behavior

The user is shown the following error message in the console.

> Required command was not provided.

The error is especially confusing because of the nature of the noun form naming of "package" and "reference". The term command is expected to refer to verbs and make it even more difficult to resolve the issue even with some familiarity with the official documentation for [dotnet-add package](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-add-package).

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.0-preview1-005977)

Product Information:
 Version:            2.0.0-preview1-005977
 Commit SHA-1 hash:  414cab8a0b

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0-preview1-005977\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview1-002111-00
  Build    : 1ff021936263d492539399688f46fd3827169983
```

**Comment by livarcocc** [comment]
This is the current behavior in the CLI:

```
Required command was not provided.
Usage: dotnet add [options] <PROJECT> [command]

Arguments:
  <PROJECT>   The project file to operate on. If a file is not specified, the command will search the current directory for one.

Options:
  -h, --help   Show help information.

Commands:
  package <PACKAGE_NAME>   .NET Add Package reference Command
  reference <args>         .NET Add Project to Project reference Command
```

It explicitly says the command is missing and then you can see below what are the possible values and formats for the commands.

**Comment by mr5z** [comment]
I'm having the same issue. Running `dotnet sln` produces similar response:
> Required command was not provided.
I have a `.sln` file on the directory I have executed this command.
I'm not sure why this issue has been closed and how the previous reply is relevant to the question.

**Comment by jpierson** [comment]
Based on the specified usage it seems that the recommended order of the `command` portion of the command is optional and suggest at the end of the expression. If I'm understanding this right my example from above would look like below.

```
> dotnet add FSharp.Data.TypeProviders package
```

I have to admit that at first I thought the ordering would be strange this way but it actually reads pretty reasonably in english. Is this order required, is the `[Command]` truly optional?

I do think there is still room for improvement in terms of these nouns being referred to as commands. If the preferred/required order is to put them at the end of the expression it doesn't seem typical based on other CLI tools which use sub commands and the two words *package* and *reference* sound more like dependency types than commands. Would it be unreasonable to refer to instead use the phrase *dependency type* or just *type* instead of *command*?

**Comment by TonyKenny666** [comment]
I'd like to +1 this because I just spent a lot of time trying to decipher this when trying to add a package to a project. It was only when I came here that I saw additional parameters which are not shown in the error message or in the help. I also assumed the [command] was optional as it's in square brackets, the same as [options].

This request:

`dotnet add ./Web Microsoft.Extensions.Options.ConfigurationExtensions`

returned the message:

```
Required command was not provided.
Usage: dotnet add [options] <PROJECT> [command]

Arguments:
  <PROJECT>   The project file to operate on. If a file is not specified, the command will search the current directory for one.

Options:
  -h, --help   Show command line help.

Commands:
  package <PACKAGE_NAME>     Add a NuGet package reference to the project.
  reference <PROJECT_PATH>   Add a project-to-project reference to the project.
```

In hindsight that I can see that I needed to add a 'command'  'package' to tell dotnet that I wanted to add a package rather than a reference.  I also could not find the documentation for this in core 3.0, maybe because it's only preview?

I agree that a 'did you mean' type of message would be more useful.
